src/routes/AppProfile/AppSettings.tsx

```diff
@@ -153,6 +154,54 @@ const AppSettings: React.FC<Props> = ({ activeProfile, setSettingsOpen }: Props)
                             }
                         </div>
                     }
+                    <div className={styles.setting}>
+                        <p>Graphics API</p>
+                        <ToggleGroup.Root
+                            className={styles.container}
+                            type="single"
+                            value={graphicsApi}
+                            onValueChange={(value) => {
+                                setGraphicsApi(value as GraphicsApi);
+                            }}>
+                            <ToggleGroup.Item className={styles.item} value={GraphicsApi.Default}>
+                                Default
+                            </ToggleGroup.Item>
+                            { (os === "windows" &&
+                                <>
+                                    <ToggleGroup.Item className={styles.item} value={GraphicsApi.D3D11}>
+                                    DirectX 11
+                                    </ToggleGroup.Item>
+                                    <ToggleGroup.Item className={styles.item} value={GraphicsApi.D3D12}>
+                                    DirectX 12
+                                    </ToggleGroup.Item>
+                                    <ToggleGroup.Item className={styles.item} value={GraphicsApi.OPEN_GL}>
+                                    OpenGL
+                                    </ToggleGroup.Item>
+                                    <ToggleGroup.Item className={styles.item} value={GraphicsApi.VULKAN}>
+                                    Vulkan
+                                    </ToggleGroup.Item>
+                                </>)
+                                || (os === "macos" &&
+                                <>
+                                    <ToggleGroup.Item className={styles.item} value={GraphicsApi.METAL}>
+                                    Metal
+                                    </ToggleGroup.Item>
+                                    <ToggleGroup.Item className={styles.item} value={GraphicsApi.VULKAN}>
+                                    Vulkan
+                                    </ToggleGroup.Item>
+                                </>)
+                                || (os === "linux" &&
+                                <>
+                                    <ToggleGroup.Item className={styles.item} value={GraphicsApi.VULKAN}>
+                                    Vulkan
+                                    </ToggleGroup.Item>
+                                    <ToggleGroup.Item className={styles.item} value={GraphicsApi.OPEN_GL}>
+                                    OpenGL
+                                    </ToggleGroup.Item>
```

Comment:
This whole part is long and has a lot of repeated entries, so consider using a dictionary or something like that to store the localized names and then convert those into a fragment of `ToggleGroup`s

Reply:
I've refactored it to avoid repetitions. As for dictionary/etc I didn't really get what you mean. I have zero prior experience with react (this is react, right?) or tauri